Parse/src/main/java/com/parse/ParseFileController.java

```diff
@@ -56,6 +56,13 @@ public File getCacheFile(ParseFile.State state) {
     return new File(cachePath, state.name());
   }
 
+  /* package for tests */ File getTempFile(ParseFile.State state) {
+    if (state.url() == null) {
+      return null;
+    }
+    return new File(cachePath, ParseDigestUtils.md5(state.url()));
```

Comment:
If we end up keeping this, we'd be clashing on the temporary file name. We'd probably want something unique like `UUID.randomUUID()`.